plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/SiriusDiagramByRepresentationAndEObjectProvider.java

```diff
@@ -0,0 +1,203 @@
+import org.eclipse.sirius.viewpoint.description.Viewpoint;
+import org.obeonetwork.m2doc.provider.DiagramProvider;
+import org.obeonetwork.m2doc.provider.IProvider;
+import org.obeonetwork.m2doc.provider.OptionType;
+import org.obeonetwork.m2doc.provider.ProviderConstants;
+import org.obeonetwork.m2doc.provider.ProviderException;
+
+/**
+ * {@link SiriusDiagramByRepresentationAndEObjectProvider} are used to get sirius diagrams from an
+ * eObject and a name.
+ * 
+ * @author Romain Guider
+ */
+public class SiriusDiagramByRepresentationAndEObjectProvider extends DiagramProvider {
+	/**
+	 * The option key used to retrieve the {@link EObject} root of some diagram representations. 
+	 */
+	private static final String KEY_TARGET_ROOT_OBJECT = "rootObject";
+	/**
+	 * The key used in the map passed to {@link IProvider} to define the Sirius
+	 * representation title from which image will be computed.
+	 */
+	private static final String KEY_DIAGRAM_DESCRIPTION_NAME = "diagramDescriptionName";
+
+	/**
+	 * Creates a new {@link SiriusDiagramByRepresentationAndEObjectProvider} given an EObject which
+	 * is the root of the seek diagram and the name of the diagram.
+	 * 
+	 * @param eObject
+	 *            the root of the diagram
+	 * @param title
+	 *            the name of the diagram
+	 * @param projectRootPath
+	 *            the project root path
+	 */
+	public SiriusDiagramByRepresentationAndEObjectProvider() {
+
+	}
+
+	/**
+	 * Replace forbidden characters with "_" in a filename
+	 * 
+	 * @param name
+	 * @return sanitized string
+	 */
+	public static String sanitizeFilename(String name) {
+		return name.replaceAll("[:\\\\/*?|<>]", "_");
+	}
+
+	/**
+	 * Get the full path (starting from the root folder) of the generated image
+	 * for a diagram
+	 * 
+	 * @param diagram
+	 * @param rootPath the project root path from which document generation has been launched.
+	 * @return Full path starting from root folder
+	 */
+	private String getDiagramImageFilename(DSemanticDiagram diagram, String rootPath) {
+		return rootPath + "/generated/images/representations/diagram_"
+				+ sanitizeFilename(diagram.eResource().getURIFragment(diagram)) + "." + ImageFileFormat.JPEG.getName();
+	}
+
+	/**
+	 * Retrieve all representations whose target is the specified EObject
+	 * 
+	 * @param object
+	 *            Object which is the target of the representations
+	 * @param session the Sirius session from which we want to find the representation with the given name.
+	 * @return List of representations
+	 */
+	private List<DRepresentation> getAssociatedRepresentationByName(EObject targetRootObject, String diagramDescriptionName, Session session) {
+		if (diagramDescriptionName == null) {
+			return new ArrayList<DRepresentation>(0);
+		}
+		List<DRepresentation> representationsResult = new ArrayList<DRepresentation>();
+		Collection<DRepresentation> representations = DialectManager.INSTANCE.getRepresentations(targetRootObject, session);
```

Comment:
If the session is null here, you'll get a NPE. That should be checked. Either here or in the caller methods.